service/controller/v24/cluster_resource_set.go

```diff
@@ -211,6 +211,7 @@ func NewClusterResourceSet(config ClusterResourceSetConfig) (*controller.Resourc
 	{
 		c := accountid.Config{
 			Logger: config.Logger,
+			STS:    config.HostAWSClients.STS,
```

Comment:
Here we set the control plane STS client which we use to fetch the account ID, which in turn is set to the controller context status.